### PR TITLE
Caret Component #2

### LINKED ISSUE #325: Collapse the sections in the Debugger sidebar
Similar to the extension manager, it would be useful to be able to collapse the different section in the debugger sidebar:

collapse-sections

### PR DESCRIPTION
**Pull request by KsavinN**
collapse
resolve #325 

Another way for collapse in splitPanel at Debugger Sidebar.

### CODE REVIEW

**Comment by jtpio** [comment]
This option looks nicer than #381 from a user's perspective :+1:

File: src/collapse.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      setNextWidgetTop(widget, nextWidget);
+      if (!nextWidget?.node.classList.contains('hide')) {
+        break;
+      }
+    }
+  }
+
+  private moveOnOpenPanel(splitWidgets: Widget[], elements: HTMLDivElement[]) {
+    let index = 0;
+    const calcutaeDiffPixel = (h1: string, h2: string) =>
```
Comment (jtpio) [inline src/collapse.ts]:
```suggestion
    const calculatePixelDiff = (h1: string, h2: string) =>
```

**Comment by jtpio** [comment]
Thanks @KsavinN for working on this.

I gave this a quick spin locally, and it looks like we can end up in situations like the following when dragging the handles of the `SplitPanel`:

collapse-sidebar-glitch

Not sure how easy it would be to prevent this (haven't looked at the logic in details yet).

**Comment by KrzysztofSikoraCodete** [comment]
Peek 2020-04-07 19-33

@jtpio I've catch second example of this problem. Looks more clearer what's going on.

**Comment by KrzysztofSikoraCodete** [comment]
panels

**Pull request closed without merging**